docs/porting-to-ROS2.md

```diff
@@ -74,7 +74,22 @@ If a package falls under all three categories (`<build_depend>`, `<build_export_
 
 
 ### Rewriting `CMakeLists.txt`
-Pretty straightforward by following the example of the already ported `simple_planning_simulator` package and the [pub-sub tutorial](https://index.ros.org/doc/ros2/Tutorials/Writing-A-Simple-Cpp-Publisher-And-Subscriber/#cpppubsub). Better yet, use `ament_auto` to get terse `CMakeLists.txt` that do not have as much redundancy with `package.xml` as an explicit `CMakeLists.txt`. See [this commit](https://github.com/tier4/Pilot.Auto/pull/7/commits/ef382a9b430fd69cb0a0f7ca57016d66ed7ef29d) for an example.
+This is not always straightforward. A starting point is to look at the [pub-sub tutorial](https://index.ros.org/doc/ros2/Tutorials/Writing-A-Simple-Cpp-Publisher-And-Subscriber/#cpppubsub). This uses `ament_cmake`, which has a [relatively good guide](https://index.ros.org/doc/ros2/Tutorials/Ament-CMake-Documentation/) that still doesn't cover everything (like what to do when installing an executable).
+
+#### `ament_cmake` and `ament_cmake_auto`
+One drawback of `ament_cmake` is that it requires typing out the dependencies at least twice, once in `package.xml` and once or more in `CMakeLists.txt`.
+
+Another possibility is to use `ament_auto` to get terse `CMakeLists.txt`. See [this commit](https://github.com/tier4/Pilot.Auto/pull/7/commits/ef382a9b430fd69cb0a0f7ca57016d66ed7ef29d) for an example. Unfortunately, there is no documentation for this tool, so you can only learn it from examples and reading the source code. It is also limited in what it does – it cannot currently generate message definitions, for instance, and always links all dependencies to all targets.
+
+There are more subtle issues too, like `ament_auto_find_build_dependencies()`. It just takes all the build dependencies verbatim from `package.xml` and calls `find_package()` without the `REQUIRED` option. This causes an issue when your library has a different name in `package.xml`/`rosdep` and in `CMake`. It also removes safety, as this won't complain when you mistype a package name or haven't installed the package yet. An alternative is to do the following:
+
+```cmake
+# By default, ament_auto_find_build_dependencies does not REQUIRE packages
+ament_auto_find_build_dependencies(REQUIRED
```

Comment:
A lot of packages don't have this explicit way of calling `ament_auto_find_build_dependencies`, should this be the new default rather than recommended?

Reply:
You mean proposing a patch to `ament_auto_find_build_dependencies()`? Or does this comment make it sound like this code snippet does not `REQUIRE` the packages? I meant it as "This is an improvement over the default (i.e. no arguments) `ament_auto_find_build_dependencies()`."

Reply:
Oh sorry about the confusion. I just meant that since it is an improvement, why would we not what it to be the default way of calling `ament_auto_find_build_dependencies`?

Reply:
Ah, yes, I would recommend that. I reworded it from an "alternative" to a "recommendation".